File: src/code-challenges/ts/towSum.ts
const twoSum = (nums: number[], target: number): number[] => {
  let result: number[] = [];
  for (let i: number = 0; i < nums.length; i++) {
    for (let j: number = 0; j < nums.length; j++) {
      if (nums[i] + nums[j] === target && i !== j) {
        console.log(i, j);
        result = [j, i];
        console.log(nums, result, [nums[result[0]], nums[result[1]]]);
        return result;
      }
    }
  }
  return result;
};

twoSum([3, 2, 4, 2], 6);

File: src/code-challenges/cpp/TowSum.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    vector<int> result;
    for(int i = 0; i <= nums.size(); i++){
        for(int j = i + 1; j <= nums.size(); j++){
            if((nums[i] + nums[j]) == target){

                result.push_back(i);
                result.push_back(j);

                return result;
            }
        }
    }
    return result;
}

int main()
{
    vector<int> arr;
    
    arr.push_back(-4);
    arr.push_back(-2);
    arr.push_back(-3);
    arr.push_back(-4);
    arr.push_back(-5);

    vector<int> funcResult = twoSum(arr,-8);
    // if(sizeof(funcResult) / 8 > 0){
        cout << funcResult[0] << endl << funcResult[1] << endl;
    // }else{
        // cout << funcResult[0] << endl;
    // }
    // bool s = (arr[2] + arr[4]) == -84; 
        // cout << arr.size() << endl;
    
    return 0;
}
